04.ls/ls_command.rb

```diff
@@ -43,4 +67,38 @@ def display_files(files)
   end
 end
 
+def display_file_details(files)
+  file_stats = files.to_h { |file| [file, File::Stat.new(file)] }
+
+  puts "total #{file_stats.values.sum(&:blocks)}"
+  print find_file_details(file_stats).join("\n")
+end
+
+def find_file_details(file_stats)
+  linksize_width = file_stats.values.map(&:nlink).max.to_s.bytesize + 2
+  filesize_width = file_stats.values.map(&:size).max.to_s.bytesize + 2
+
+  file_stats.map do |file, file_stat|
+    cols = []
+    cols << FILE_TYPE_CHARACTER[file_stat.ftype]
+    cols << convert_file_permissions(file_stat)
+    cols << file_stat.nlink.to_s.rjust(linksize_width)
+    cols << " #{Etc.getpwuid(file_stat.uid).name}  #{Etc.getgrgid(file_stat.gid).name}"
```

Comment:
```suggestion
    cols << " #{Etc.getpwuid(file_stat.uid).name}"
    cols << "  #{Etc.getgrgid(file_stat.gid).name}"
```

2列分のデータなので2回に分けた方が自然かなと思いました